addons/im_livechat/static/tests/chat_window_patch.test.js

```diff
@@ -29,3 +31,55 @@ test("closing a chat window with no message from admin side unpins it", async ()
     });
     await contains(".o_notification", { text: "You unpinned your conversation with Demo" });
 });
+
+test("Focus should not be stolen when a new livechat open", async () => {
+    const pyEnv = await startServer();
+    const [guestId1, guestId2] = pyEnv["mail.guest"].create([
+        { name: "Visitor 11" },
+        { name: "Visitor 12" },
+    ]);
+    const channelIds = pyEnv["discuss.channel"].create([
+        {
+            anonymous_name: "Visitor 11",
+            channel_member_ids: [
+                Command.create({
+                    partner_id: serverState.partnerId,
+                    last_interest_dt: "2021-01-03 10:00:00",
+                }),
+                Command.create({ guest_id: guestId1, last_interest_dt: "2021-01-03 10:00:00" }),
+            ],
+            channel_type: "livechat",
+            livechat_operator_id: serverState.partnerId,
+        },
+        {
+            anonymous_name: "Visitor 12",
+            channel_member_ids: [
+                Command.create({
+                    partner_id: serverState.partnerId,
+                    last_interest_dt: "2021-01-03 10:00:00",
+                }),
+                Command.create({ guest_id: guestId2, last_interest_dt: "2021-01-03 10:00:00" }),
+            ],
+            channel_type: "livechat",
+            livechat_operator_id: serverState.partnerId,
+        },
+    ]);
+    await start();
+    await click(".o_menu_systray i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem", { text: "Visitor 11" });
+    await contains(".o-mail-ChatWindow", { text: "Visitor 11" });
+    await contains(".o-mail-Composer-input[placeholder='Message Visitor 11…']:focus");
+    withGuest(guestId2, () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "hu",
+                message_type: "comment",
+                subtype_xmlid: "mail.mt_comment",
+            },
+            thread_id: channelIds[1],
+            thread_model: "discuss.channel",
+        })
+    );
+    await contains(".o-mail-ChatWindow", { text: "Visitor 12" });
+    await contains(".o-mail-ChatWindow:nth-child(2) .o-mail-Composer-input:focus");
```

Comment:
```suggestion
    await contains(".o-mail-Composer-input[placeholder='Message Visitor 11…']:focus");
    await contains(".o-mail-Composer-input[placeholder='Message Visitor 12…']:focus", { count: 0 });
```

Reply:
Sadly no, because sometimes, the placeholder is replaced by something like "use tab to switch chat window", it's unreliable here.

Reply:
I also think we should have the same assert before and after.

This relies on chat window ordering which while it is supposed to be correct, we don't want the test to depend on.

Reply:
But it is a negative assert test so value is very low. At least add an await animation frame before the final assert to give more time to potential side effects to happen.